Data.Map.fold has been deprecated in containers 0.6

https://github.com/jgoerzen/hslogger/blob/9368275658d465e0738247dfeec4233021c877b4/src/System/Log/Logger.hs#L479

I get the following error when compiling:

```
src/System/Log/Logger.hs:479:27: error:
    • Data.Map.fold is gone. Use foldr.
    • In the expression:
        Map.fold (\ l r -> concat [r, handlers l]) [] lt
      In an equation for ‘allHandlers’:
          allHandlers = Map.fold (\ l r -> concat [r, handlers l]) [] lt
      In the expression:
        do let allHandlers = Map.fold (\ l r -> ...) ... lt
           mapM_ (\ (HandlerT h) -> close h) allHandlers
           return $ Map.map (\ l -> l {handlers = []}) lt
    |
479 |         let allHandlers = Map.fold (\l r -> concat [r, handlers l]) [] lt
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Failed to install hslogger-1.2.10
```

The fix seems to be trivial: just replacing `fold` with `foldr` as the error message advices, but I'm not sure about backward compatibility.

--- a/src/System/Log/Logger.hs
+++ b/src/System/Log/Logger.hs
@@ -476,7 +476,7 @@ updateGlobalLogger ln func =
 removeAllHandlers :: IO ()
 removeAllHandlers =
     modifyMVar_ logTree $ \lt -> do
-        let allHandlers = Map.fold (\l r -> concat [r, handlers l]) [] lt
+        let allHandlers = Map.foldr (\l r -> concat [r, handlers l]) [] lt
         mapM_ (\(HandlerT h) -> close h) allHandlers
         return $ Map.map (\l -> l {handlers = []}) lt
 

